File: drills/elixir/solutions/nc38_meeting_rooms__pairwise.exs
defmodule Solution do
  # Every pair, checked. Two intervals overlap when each starts before the other
  # ends -- the condition worth being able to write from memory, since it is
  # easier to get right than its negation.
  def can_attend_meetings?(intervals) do
    pairs =
      for {a, i} <- Enum.with_index(intervals),
          {b, j} <- Enum.with_index(intervals),
          i < j,
          do: {a, b}

    Enum.all?(pairs, fn {{s1, e1}, {s2, e2}} -> not (s1 < e2 and s2 < e1) end)
  end
end

File: drills/go/solutions/nc38_meeting_rooms__pairwise.go
package main

func canAttendMeetings(intervals [][]int) bool {
	for i := 0; i < len(intervals); i++ {
		for j := i + 1; j < len(intervals); j++ {
			if intervals[i][0] < intervals[j][1] && intervals[j][0] < intervals[i][1] {
				return false
			}
		}
	}
	return true
}
